module2/src/com/epam/decomposition/LogicBuilder.java

```diff
@@ -0,0 +1,93 @@
+package com.epam.decomposition;
+
+import com.epam.sorting.ArraySorter;
+
+public class LogicBuilder {
+
+    //task 1
+    public static int findLeastCommonMultiple(int firstNumber, int secondNumber) {
+        return ((firstNumber * secondNumber) / (SupportingLogicalBuilder.
+                findGreatestCommonDivisor(firstNumber, secondNumber)));
+    }
+
+    //task 2
+    public static int findLeastCommonMultipleForFourNumbers(int firstNumber, int secondNumber,
+                                                             int thirdNumber, int fourthNumber) {
+        return SupportingLogicalBuilder.findGreatestCommonDivisor((SupportingLogicalBuilder.
+                findGreatestCommonDivisor(firstNumber, secondNumber)),
+                (SupportingLogicalBuilder.findGreatestCommonDivisor(thirdNumber, fourthNumber)));
+    }
+
+    //task 3
+    public static double findHexagonArea(double sizeHexagon, double amountOfCorners) {
+        return SupportingLogicalBuilder.calculateEquilateralTriangleArea(sizeHexagon) * amountOfCorners;
+    }
+
+    //task 4
+    public static double findGreatestDistance(double[] array) {
+        double maxDistance = 0;
+        for (double i = 0; i < array.length; i++) {
+            for (double j = 1; j < array.length - j; j++) {
+                if ((SupportingLogicalBuilder.
+                        calculatePointSpacing(i ,i ,i + j,i + j)) > maxDistance) { maxDistance =
+                        (SupportingLogicalBuilder.
+                            calculatePointSpacing(i, i,  i + j, i + j));
+                }
+            }
+        }
+        return maxDistance;
+    }
```

Comment:
please move at least 3-4 tasks to separate class, let's say GeometricalCalculator